crates/core/src/config/mod.rs

```diff
@@ -64,15 +64,28 @@ pub struct Config {
 }
 
 impl Config {
+    /// Returns a `PathBuf` to the flavor directory.
+    pub fn get_flavor_directory_for_flavor(&self, flavor: &Flavor, path: &PathBuf) -> PathBuf {
+        path.join(&flavor.folder_name())
+    }
+
+    /// Returns a `Option<PathBuf>` to the root directory of the Flavor.
+    pub fn get_root_directory_for_flavor(&self, flavor: &Flavor) -> Option<PathBuf> {
+        if let Some(flavor_dir) = self.wow.directories.get(flavor) {
+            Some(Box::new(flavor_dir.parent().unwrap()).to_path_buf())
```

Comment:
Is the box necessary here?

Reply:
Mainly to [convert Path to PathBuf](https://doc.rust-lang.org/std/path/struct.Path.html#see-also-2). Do you want to do it in another way?

Reply:
Path already has a function `to_path_buf`, the box is unnecessary https://doc.rust-lang.org/std/path/struct.Path.html#method.to_path_buf